spring-integration-mongodb/src/test/java/org/springframework/integration/mongodb/metadata/MongoDbMetadataStoreTests.java

```diff
@@ -0,0 +1,136 @@
+import org.junit.Test;
+import org.springframework.data.mongodb.MongoDbFactory;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.integration.mongodb.rules.MongoDbAvailable;
+import org.springframework.integration.mongodb.rules.MongoDbAvailableTests;
+
+
+/**
+ * @author Senthil Arumugam, Samiraj Panneer Selvam
+ *
+ */
+public class MongoDbMetadataStoreTests extends MongoDbAvailableTests {
+	
+	private final static String DEFAULT_COLLECTION_NAME = "metadatastore";
+	private final String file1 = "/remotepath/filesTodownload/file-1.txt";
+	private final String file1Id = "12345";
+	private MongoDbMetadataStore store = null;
+	
+	@MongoDbAvailable
+	@Before
+	public void configure() throws Exception  {
+		MongoDbFactory mongoDbFactory = this.prepareMongoFactory(DEFAULT_COLLECTION_NAME);
+		MongoTemplate template = new MongoTemplate(mongoDbFactory);
+		store = new MongoDbMetadataStore(template);
+	}
+	
+	@MongoDbAvailable
+	@Test
+	public void testConfigureCustomCollection() throws Exception  {
+		String collectioName = "TestMetaStore";
+		MongoDbFactory mongoDbFactory = this.prepareMongoFactory(collectioName);
+		MongoTemplate template = new MongoTemplate(mongoDbFactory);
+		store = new MongoDbMetadataStore(template,collectioName);
+		testBasics();
+	}
+	
+	@MongoDbAvailable
+	@Test
+	public void testConfigureFactory() throws Exception  {
+		MongoDbFactory mongoDbFactory = this.prepareMongoFactory(DEFAULT_COLLECTION_NAME);
+		store = new MongoDbMetadataStore(mongoDbFactory);
+		testBasics();
+	}
+	
+	@MongoDbAvailable
+	@Test
+	public void testConfigureFactorCustomCollection() throws Exception  {
+		String collectioName = "TestMetaStore";
+		MongoDbFactory mongoDbFactory = this.prepareMongoFactory(collectioName);
+		store = new MongoDbMetadataStore(mongoDbFactory,collectioName);
+		testBasics();
+	}
+	
+	private void testBasics(){
+		String fileID = store.get(file1);
+		assertNull(fileID);
+		
+		store.put(file1, file1Id);
+		
+		fileID = store.get(file1);
+		assertNotNull(fileID);
+		assertEquals(file1Id, fileID);
+	}
+	
+	@Test
+	@MongoDbAvailable
+	public void testGetFromStore(){
+		testBasics();
+	}
+	
+	@Test
+	@MongoDbAvailable
+	public void testPutIfAbsent() throws Exception{
+		String fileID = store.get(file1);
+		assertNull(fileID);
+		
+		fileID = store.putIfAbsent(file1, file1Id);
+		assertNull(fileID);
+		
+		fileID = store.putIfAbsent(file1, file1Id);
+		assertNotNull(fileID);
+		assertEquals(file1Id, fileID);
+	}
+	
+	@Test
+	@MongoDbAvailable
+	public void testRemove() throws Exception{
+		String fileID = store.remove(file1);
+		assertNull(fileID);
+		
+		fileID = store.putIfAbsent(file1, file1Id);
+		assertNull(fileID);
+		
+		fileID = store.remove(file1);
+		assertNotNull(fileID);
+		assertEquals(file1Id, fileID);
+	}
+	
+	@Test
+	@MongoDbAvailable
+	public void testReplace() throws Exception{
+		boolean removedValue = store.replace(file1, file1Id, "4567");
+		assertFalse(removedValue);
+		
```

Comment:
I think here must be ` store.get(file1)` to be sure that the `replace` hasn't affected the collection.